### PR TITLE
projects: Add coala_GUI.md

### PR DESCRIPTION
**Pull request by HitRam**
Project proposal for coala-GUI. 

PS: First project proposal.

### CODE REVIEW

File: data/projects.js [hunk 1]
```diff
@@ -361,5 +361,20 @@ coala should support generating metrics for your code.",
...
 		"markdown": "improve_installation.md",
 		"collaborating_projects" : ["coala"]
-	}
+	},
+	{
+		"name" : "Create coala-GUI",
+		"desc" : "Crate a GUI for coala so that the user can analyse the \
+		 code in the most easiest way without worrying about any more commands.",
+		"requirements" : [
+				"The proposal needs to contain a clear concept"
```
Comment (sims1253) [inline data/projects.js]: mockups, gui experience?

File: data/projects/coala_GUI.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+The main aim of this project is to create a GUI for coala to make code Analysis even easier for users. A GUI would be the easiest way for users to get familiar with coala and it's functionalites. With more and more bears and other functionalites that may be added to coala, the number of commands will increase, and hence the user has to get more accustomed. Because GUI is much more visually intutive as compared to CLI, it would be very easy for user to explore coala's functionalities without worrying about any commands. This project basically requires intergration of current features of coala into coala GUI.
```
Comment (sims1253) [inline data/projects/coala_GUI.md]: consider limiting line length to 80 ;)

File: data/projects/coala_GUI.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+* The user can view/edit/run .coafile with options of executing single/multiple sections manually.
+* GUI can suggest best suited configuration file for the selected files (Basically integration of coala-GUI with coala-quickstart)
+
+Apart from above mentioned features, it depends on applicant's creativity to add new features.
+
+#### Milestones
+
+#### GSOC 2017 CODING STARTS
+
+* The applicant is familiar with core coala codebase.
```
Comment (sims1253) [inline data/projects/coala_GUI.md]: depending on how you define familiar only the oldest and most experienced maintainers would pass this ;)

File: data/projects/coala_GUI.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+* The applicant is familiar with the library which is to be used (preferably PyQt5).
```
Comment (sims1253) [inline data/projects/coala_GUI.md]: for consistend ui across OSs you might consider some web ui thingy

File: data/projects/coala_GUI.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+* Basic layout of GUI is ready.
```
Comment (sims1253) [inline data/projects/coala_GUI.md]: Does coala offer endpoints for a ui to use? Do we need api changes(probably). The concept of how to connect coala with the gui should be clear.

File: data/projects/coala_GUI.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+#### GSOC 2017 MIDTERM
+
+* The GUI is working with minimum functionalities. (user is able to load files and select the required bears and gets option for atleast one output format).
+* The GUI is able to execute sections and display its results in a specific format.
+
+#### GSOC 2017 FINAL
+
+* The GUI is fully integrated with coala-quickstart (user has the option of generating automatic configuration file).
+* The GUI is fully functional with mentioned features.
```
Comment (sims1253) [inline data/projects/coala_GUI.md]: Everything needs in depth documentation and testing

File: data/projects.js [hunk 7]
```diff
@@ -361,5 +361,20 @@ coala should support generating metrics for your code.",
...
+		],
+		"difficulty" : "medium/high",
```
Comment (sims1253) [inline data/projects.js]: voting for high

**Comment by sims1253** [comment]
could you also formulate why coala needs/wants a gui in a nice paragraph? I guess I know your Idea behind it but considering who the target audience of coala is(devs) why would a gui be a nice thing?

File: data/projects/coala_GUI.md (hunk 4 above)
Comment (HitRam) [inline data/projects/coala_GUI.md]: PyQt is cross platform and works well. Other option is electron but it's performance is a bit slow as compared to PyQt and hence I mentioned that.

**Comment by HitRam** [comment]
@sims1253  Yes I'll.
I actually wanted to take some time and work on the idea but I was told to submit the project proposal asap.

File: data/projects.js [hunk 8]
```diff
@@ -361,5 +361,20 @@ coala should support generating metrics for your code.",
...
+		"issues" : [],
+		"mentors" : ["@Udayan12167", "@sils"],
```
Comment (sils) [inline data/projects.js]: I'm not mentoring this and I'm voting against this project happening at all. coala is already fragmented a lot and coala-html serves a similar purpose, we should focus on that one indeed. Not trying to be mean but sorry, I'm totally against this.

**Comment by sils** [comment]
unack https://github.com/coala/projects/pull/133/commits/372e25a024c9489d48b2dc56a49b4ea044112e7d from me until someone convinces me that there's a *real* need for a GUI

**Comment by Adrianzatreanu** [comment]
I disagree with the need of a coalaGUI too, as long as we have a prototype for coala-html. I think coala-html is way too good of an idea for it not be continued and taken care of.

**Comment by sims1253** [comment]
I feel like good IDE plugins would fill the role this could play.
Closing this as it sounds rejected to me.

**Pull request closed without merging**